js/src/dom/data.js

```diff
@@ -11,57 +11,44 @@
  * ------------------------------------------------------------------------
  */
 
-const mapData = (() => {
-  const storeData = {}
-  let id = 1
-  return {
-    set(element, key, data) {
-      if (typeof element.bsKey === 'undefined') {
-        element.bsKey = {
-          key,
-          id
-        }
-        id++
-      }
+const elementMap = new Map()
 
-      storeData[element.bsKey.id] = data
-    },
-    get(element, key) {
-      if (!element || typeof element.bsKey === 'undefined') {
-        return null
-      }
+export default {
+  set(element, key, instance) {
+    if (!elementMap.has(element)) {
+      elementMap.set(element, new Map())
+    }
 
-      const keyProperties = element.bsKey
-      if (keyProperties.key === key) {
-        return storeData[keyProperties.id]
-      }
+    const instanceMap = elementMap.get(element)
 
-      return null
-    },
-    delete(element, key) {
-      if (typeof element.bsKey === 'undefined') {
-        return
-      }
-
-      const keyProperties = element.bsKey
-      if (keyProperties.key === key) {
-        delete storeData[keyProperties.id]
-        delete element.bsKey
-      }
+    // make it clear we only want one instance per element
+    // can be removed later when multiple key/instances are fine to be used
+    if (!instanceMap.has(key) && instanceMap.size !== 0) {
+      // eslint-disable-next-line no-console
+      console.error(`Notice: The design of Bootstrap currently discourages to use more than one instance per element. Bound instance(s): ${[...instanceMap.keys()].join(', ')}.`)
```

Comment:
I get `[object Map Iterator]` instead the keys' names in the console :thinking: 
![image](https://user-images.githubusercontent.com/26922866/108537456-a11c1380-7303-11eb-8bb0-285a520df91d.png)

Reply:
Mh looks like that's a problem of how the code is processed.

In the source there is `[...instanceMap.keys()].join(', ')` with the spread operator taking care of `keys()` returning an iterator object and iterates that automatically. While in the composed script it is replaced with `[].concat(instanceMap.keys()).join(', ')` which concats that iterator object into an array without iterating.

Replacing it with `Array.from()` to solve that problem.